plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1540 @@
+                    if matched_user:
+                        matched_users.extend(matched_user)
+                        user_url = "data/ietf-snmp:snmp/usm/local/user={}".format(matched_user['name'])
+                        user_request = {"path": user_url, "method": DELETE}
+                        user_requests.append(user_request)
+
+        return user_requests
+
+    def same_options(self, options_1, options_2):
+        """ Determines if the given list of options have the same values
+
+        :rtype: Boolean
+        :returns: True if the values are the same
+        """
+        for key1, value1 in options_1.items():
+            for key2, value2 in options_2.items():
+                if key1 == key2:
+                    if value1 != value2:
+                        return False
+        return True
+
+    def get_delete_snmp_request(self, configs, have, delete_all):
+        """ Create the requests necessary to delete the given configuration
+        :rtype: A list
+        :returns: The list of requests to delete the users in "configs" from the current configuration
+        """
+        requests = []
+        if not configs:
+            return requests
+
+        agentaddress_requests_list = []
+        community_requests_list = []
+        contact_requests_list = []
+        enable_trap_requests_list = []
+        engine_requests_list = []
+        group_requests_list = []
+        host_requests_list = []
+        location_requests_list = []
+        user_requests_list = []
+        view_requests_list = []
+
+        agentaddress = 'agentaddress' in configs
+        community = 'community' in configs
+        contact = 'contact' in configs
+        enable_trap = 'enable_trap' in configs
+        engine = 'engine' in configs
+        group = 'group' in configs
+        host = 'host' in configs
+        location = 'location' in configs
+        user = 'user' in configs
+        view = 'view' in configs
+
+        have_agentaddress = have.get('agentaddress')
+        have_community = have.get('community')
+        have_contact = have.get('contact')
+        have_enable_trap = have.get('enable_trap')
+        have_engine = have.get('engine')
+        have_group = have.get('group')
+        have_host = have.get('host')
+        have_location = have.get('location')
+        have_user = have.get('user')
+        have_view = have.get('view')
+
+        if delete_all or agentaddress:
+            if have_agentaddress is not None:
+                agentaddress_requests = []
+                if delete_all or self.check_dicts_matched(have_agentaddress, configs['agentaddress']):
+                    agentaddress_url = "data/ietf-snmp:snmp/engine/listen"
+                    agentaddress_request = {"path": agentaddress_url, "method": DELETE}
+                    agentaddress_requests.append(agentaddress_request)
+                else:
+                    after_deletion = get_diff(configs['agentaddress'], have_agentaddress)
```

Comment:
The order of get_diff arguments is reversed here. Instead, it should be:

```suggestion
                    after_deletion = get_diff(have_agentaddress, configs['agentaddress'])
```